src/test/java/me/hydos/rosella/example/fboWater/FboWaterTest.java

```diff
@@ -0,0 +1,213 @@
+package me.hydos.rosella.example.fboWater;
+
+import me.hydos.rosella.Rosella;
+import me.hydos.rosella.display.GlfwWindow;
+import me.hydos.rosella.example.source.GenericSourceTest;
+import me.hydos.rosella.file.model.GlbModelLoader;
+import me.hydos.rosella.file.model.GlbRenderObject;
+import me.hydos.rosella.render.PolygonMode;
+import me.hydos.rosella.render.Topology;
+import me.hydos.rosella.render.material.Material;
+import me.hydos.rosella.render.model.GuiRenderObject;
+import me.hydos.rosella.render.pipeline.Pipeline;
+import me.hydos.rosella.render.pipeline.state.StateInfo;
+import me.hydos.rosella.render.resource.Global;
+import me.hydos.rosella.render.resource.Identifier;
+import me.hydos.rosella.render.resource.Resource;
+import me.hydos.rosella.render.shader.RawShaderProgram;
+import me.hydos.rosella.render.shader.ShaderProgram;
+import me.hydos.rosella.render.texture.*;
+import me.hydos.rosella.render.vertex.VertexFormats;
+import me.hydos.rosella.scene.object.impl.SimpleObjectManager;
+import me.hydos.rosella.test_utils.NoclipCamera;
+import me.hydos.rosella.util.Color;
+import org.joml.Matrix4f;
+import org.joml.Vector3f;
+import org.lwjgl.system.Configuration;
+import org.lwjgl.vulkan.VK10;
+
+import java.util.List;
+
+/**
+ * Test which contains source related stuff
+ */
+public class FboWaterTest {
```

Comment:
maybe make a basic test class that has these methods, but abstract to make all tests have a similar structure?

Reply:
not for this pr irr

Reply:
sure that can be later. just something I noticed